rfcs/0001-messages.md

```diff
@@ -0,0 +1,314 @@
+
+`Payload`s use data types that we need to specify first. We have 2 primitive types inside the payload messages:
+
+#### `Scalar`
+
+`Scalar` is a an alias for `jubjub::Fr` and this is a `[u64; 4]` as documented in https://github.com/zkcrypto/jubjub/blob/main/src/fr.rs#L16
+
+#### `AffinePoint`
+
+Much of the math in FROST is done using `jubjub::ExtendedPoint`. This is a structure with 5 `jubjub::Fq`s as defined in https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L128-L134
+
+Each `Fq` needed to form a `jubjub::ExtendedPoint` are `Scalar`s of `bls12_381` crate. Scalar here is `[u64; 4]` as documented in https://github.com/zkcrypto/bls12_381/blob/main/src/scalar.rs#L16
+
+For message exchange `jubjub::AffinePoint`s are a better choice as they are shorter in bytes, they are formed of 2 `jubjub::Fq` instead of 5: https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L70-L73
+
+Conversion from one type to the other is trivial:
+
+https://docs.rs/jubjub/0.6.0/jubjub/struct.AffinePoint.html#impl-From%3CExtendedPoint%3E
+https://docs.rs/jubjub/0.6.0/jubjub/struct.ExtendedPoint.html#impl-From%3CAffinePoint%3E
+
+### Payload
+
+Payload part of the message is variable in size and depends on message type.
+
+#### `MsgDealerBroadcast`
+
+Bytes  | Field name  | Data type
+-------|-------------|-----------
+256    | secret_key  | Scalar
+512    | commitments | AffinePoint
+512+32 | group_public| GroupPublic
+
+#### `MsgCommitments`
+
+Bytes   | Field name | Data type
+--------|------------|-----------
+512+512 | commitment | Commitment
+
+#### `MsgSigningPackage`
+
+Bytes                  | Field name     | Data type
+-----------------------|----------------|-----------
+1                      | participants   | u8
+(1+1024)*partipants    | commitments    | Vec<CollectedCommitments>
+8                      | message_length | u64
+message_length         | message        | [u8]
+
+#### `SignatureShare`
+
+Bytes | Field name | Data type
+------|------------|-----------
+256   | signature  | Scalar
+
+#### `MsgFinalSignature`
+
+Bytes | Field name       | Data type
+------|------------------|-----------
+64    | final_signature  | FinalSignature
+
+## Not included
+
+The following are a few things this RFC is not considering:
+
+- After the dealer sends the initial `MsgDealerBroadcast` to all the participants, the aggregator must wait for signers to send the second message `MsgCommitments`. There is no timeout for this but only after the aggregator received all the commitments the process can continue. These restrictions and event waiting are not detailed in this RFC.
+- This implementation considers not only communications between computer devices in the internet but allows the process to be done by other channels, the lack of timers can result in participants waiting forever for a message. It is the participants business to deal with this and other similars.
+- The RFC does not describe a Service but just message structure and serialization.
+
```

Comment:
Shouldn't we limit the size of the messages to at most 4 GB by using u32 instead of u64? Using u64 allows the participants to set intractable lengths.